src/ui/public/filter_bar/lib/__tests__/map_geo_polygon.js

```diff
@@ -0,0 +1,82 @@
+import expect from 'expect.js';
+import ngMock from 'ng_mock';
+import { FilterBarLibMapGeoPolygonProvider } from 'ui/filter_bar/lib/map_geo_polygon';
+
+describe('Filter Bar Directive', function () {
+  describe('mapGeoPolygon()', function () {
+    let mapGeoPolygon;
+    let $rootScope;
+
+    beforeEach(ngMock.module(
+      'kibana',
+      'kibana/courier',
+      function ($provide) {
+        $provide.service('courier', require('fixtures/mock_courier'));
+      }
+    ));
+
+    beforeEach(ngMock.inject(function (Private, _$rootScope_) {
+      mapGeoPolygon = Private(FilterBarLibMapGeoPolygonProvider);
+      $rootScope = _$rootScope_;
+    }));
+
+    it('should return the key and value for matching filters with bounds', function (done) {
+      const filter = {
+        meta: {
+          index: 'logstash-*'
+        },
+        geo_polygon: {
+          point: { // field name
+            points: [
+              { lat: 5, lon: 10 },
+              { lat: 15, lon: 20 }
+            ]
+          }
+        }
+      };
+      mapGeoPolygon(filter).then(function (result) {
+        expect(result).to.have.property('key', 'point');
+        expect(result).to.have.property('value');
+        // remove html entities and non-alphanumerics to get the gist of the value
+        expect(result.value.replace(/&[a-z]+?;/g, '').replace(/[^a-z0-9]/g, '')).to.be('lat5lon10lat15lon20');
+        done();
+      });
+      $rootScope.$apply();
+    });
+
+    it('should return undefined for none matching', function (done) {
+      const filter = { meta: { index: 'logstash-*' }, query: { query_string: { query: 'foo:bar' } } };
+      mapGeoPolygon(filter).catch(function (result) {
```

Comment:
Honestly, this whole file is a copy/paste job from the other mapper tests. I pretty much never use `done` myself. While I think your suggestion is a good one, I think it would make more sense to apply this as a refactoring to all of the mapper tests in a separate PR if we think it's a worthwhile change.